src/Common/src/System/LocalAppContext.cs

```diff
@@ -35,29 +36,12 @@ private static bool GetCachedSwitchValueInternal(string switchName, ref int swit
             return isSwitchEnabled;
         }
 
-        private static bool DisableCaching 
-        { 
-            get 
+        private static bool DisableCaching =>
+            LazyInitializer.EnsureInitialized(ref s_disableCaching, ref s_isDisableCachingInitialized, ref s_syncObject, () =>
```

Comment:
Any time a lambda/delegate closes over instance state, including implicitly over `this`, you'll get at least one allocation for the delegate, and if more than `this` is needed, you'll also get a closure object to store all of the state.  If the lambda/delegate only ever accesses static state, then the compiler can avoid any closure allocation and can generate code that lazily-initializes a delegate for the call site, caching it for subsequent access.